compiler/rustc_ast_passes/src/feature_gate.rs

```diff
@@ -276,27 +276,6 @@ impl<'a> PostExpansionVisitor<'a> {
             );
         }
     }
-
-    /// Feature gate `impl Trait` inside `type Alias = $type_expr;`.
-    fn check_impl_trait(&self, ty: &ast::Ty) {
-        struct ImplTraitVisitor<'a> {
-            vis: &'a PostExpansionVisitor<'a>,
-        }
-        impl Visitor<'_> for ImplTraitVisitor<'_> {
-            fn visit_ty(&mut self, ty: &ast::Ty) {
-                if let ast::TyKind::ImplTrait(..) = ty.kind {
-                    gate_feature_post!(
-                        &self.vis,
-                        min_type_alias_impl_trait,
```

Comment:
We should not remove the feature gate check, instead it should check `type_alias_impl_trait` now, otherwise this PR insta-stabilizes min_tait ^^